src/L2/discounts/ERC1155DiscountValidatorV2.sol

```diff
@@ -47,10 +50,17 @@ contract ERC1155DiscountValidatorV2 is IDiscountValidator {
         uint256[] memory ids = abi.decode(validationData, (uint256[]));
         for (uint256 i; i < ids.length; i++) {
             uint256 id = ids[i];
-            if (approvedTokenIds[id] && token.balanceOf(claimer, id) > 0) {
+            if (approvedTokenIds[id] && _getBalance(claimer, id) > 0) {
                 return true;
             }
         }
         return false;
     }
+
+    /// @notice Helper for staticcalling getBalance to avoid reentrancy vector.
+    function _getBalance(address claimer, uint256 id) internal view returns (uint256) {
```

Comment:
This is great. I like that you pulled in the OZ Address library for this instead of writing a new implementation.